numba/core/config.py

```diff
@@ -15,8 +15,11 @@
 
 import llvmlite.binding as ll
 
+_uname = platform.uname()
+
 IS_WIN32 = sys.platform.startswith('win32')
 IS_OSX = sys.platform.startswith('darwin')
+IS_OSX_ARM64 = _uname.system == 'Darwin' and _uname.machine == 'arm64'
```

Comment:
Thinking about the impact this has on import time... does this need to be in here opposed to in e.g. `numba.tests.support` ? `uname` will make a call to `uname(3P)` and IIRC `core/config.py` is on the main import path from `__init__.py`.

Reply:
I'd also rather they were all in one place (e.g. `numba.core.system` or something), and also lazy, but that's out of scope for this PR. As this check is solely used in `numba.tests.support`, I think it's ok to move it there, there are similar system checks there already.